src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -468,23 +473,32 @@ namespace winrt::TerminalApp::implementation
     //   configurations. See CascadiaSettings::BuildSettings for more details.
     void TerminalPage::_OpenNewTab(const winrt::TerminalApp::NewTerminalArgs& newTerminalArgs)
     {
-        const auto [profileGuid, settings] = _settings->BuildSettings(newTerminalArgs);
-
-        _CreateNewTabFromSettings(profileGuid, settings);
-
-        const int tabCount = static_cast<int>(_tabs.size());
-        const bool usedManualProfile = (newTerminalArgs != nullptr) &&
-                                       (newTerminalArgs.ProfileIndex() != nullptr ||
-                                        newTerminalArgs.Profile().empty());
-        TraceLoggingWrite(
-            g_hTerminalAppProvider, // handle to TerminalApp tracelogging provider
-            "TabInformation",
-            TraceLoggingDescription("Event emitted upon new tab creation in TerminalApp"),
-            TraceLoggingInt32(tabCount, "TabCount", "Count of tabs curently opened in TerminalApp"),
-            TraceLoggingBool(usedManualProfile, "ProfileSpecified", "Whether the new tab specified a profile explicitly"),
-            TraceLoggingGuid(profileGuid, "ProfileGuid", "The GUID of the profile spawned in the new tab"),
-            TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
-            TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance));
+        try
+        {
+            const auto [profileGuid, settings] = _settings->BuildSettings(newTerminalArgs);
+
+            _CreateNewTabFromSettings(profileGuid, settings);
+
+            const int tabCount = static_cast<int>(_tabs.size());
+            const bool usedManualProfile = (newTerminalArgs != nullptr) &&
+                                           (newTerminalArgs.ProfileIndex() != nullptr ||
+                                            newTerminalArgs.Profile().empty());
+            TraceLoggingWrite(
+                g_hTerminalAppProvider, // handle to TerminalApp tracelogging provider
+                "TabInformation",
+                TraceLoggingDescription("Event emitted upon new tab creation in TerminalApp"),
+                TraceLoggingInt32(tabCount, "TabCount", "Count of tabs curently opened in TerminalApp"),
+                TraceLoggingBool(usedManualProfile, "ProfileSpecified", "Whether the new tab specified a profile explicitly"),
+                TraceLoggingGuid(profileGuid, "ProfileGuid", "The GUID of the profile spawned in the new tab"),
+                TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance));
+        }
+        CATCH_LOG();
+        // TODO: Should we display a dialog when we do nothing because we
+        // couldn't find the associated profile? This can only really happen in
+        // the duplicate tab scenario currently, but maybe in the future of
+        // keybindings with args, someone could manually open a tab/pane with
+        // specific a GUID.
```

Comment:
Should we actually duplicate a tab just by pushing its control's TerminalSettings object into another control? That might be even easier than a dialog, and more "expected"

Reply:
So the related case is that of when you try to `{action:"newTab", "profile":"{guid-of-profile-that-doesnt-exist}"}` - in `TerminalPage::_OpenNewTab` - in _that_ case should there be a dialog?

Reply:
Also that might be trickier. `TermControl` doesn't currently expose a way to retrieve it's `Settings` currently, nor does the `Pane` hang on to that `Settings` object after the control is initialized.

Reply:
@DHowett-MSFT @cinnamon-msft necro'ing this PR. Lets come to a consensus, what should we do when the user tries to duplicate a tab for a tab with a profile that doesn't exist? 

1. I actually don't hate the idea of just duplicating the settings for the existing control. In that scenario, we'd probably just ignore the settings for that profile in the actual file, and just use whatever settings the control currently has. This might require some plumbing, but it's doable.
2. alternatively, we could display some sort of dialog to the user, though I'd want PM input on what that text should say.
3. Scenario 3, which is what I've currently got in this PR is "just silently do nothing".

Reply:
I don't mind option 1 here. It's still what the user would expect. 😄

Reply:
1 it is then :) I'll loop back around on this one (eventually)

Reply:
yeah i love 1

Reply:
Oooh this is actually much harder.

The connection _type_ is not a part of the `IControlSettings`, it's something that's instantiated separately. So we can get the control settings back out of the control for a control that doesn't have a valid profile anymore, but we won't be able to generate a new connection for that control.

Reply:
Okay this is harder, but maybe not impossible, but probably impossible in v1.

The `Profile` knows how to build the connection settings, and the `ControlSettings`, etc. However, when the settings reloads like in this bug, we'll remove all the old profiles, and build a new list. 

If we wanted the case 1 behavior, we could wait until the `Profile` is a winrt object, then have each Pane hold a strong ref to the `Profile` that it's hosting. Then it would be trivial to be able to duplicate the settings from it. I believe that work is tracked in #3998.

How palatable is doing case 3 now (to fix the crash), then filing a follow up to do case 1 once #3998 is done?

Reply:
Jeeeeez, okay, let's do case 3.

Reply:
Moved to #5047.